sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/TracerProxy.java

```diff
@@ -84,11 +86,9 @@ public static void end(int responseCode, Throwable error, Context context) {
      * @return An updated {@link Context} object.
      */
     public static Context setSpanName(String spanName, Context context) {
```

Comment:
Why does this API return `Context`? It's inconsistent with `setAttribute()` API which returns a void. Or, should setAttribute() return Context too since `start()` is also returning `Context`?

Reply:
`start()` should return a `Context` as we update the context to hold the span that it just created/started.
`setAttribute()`  shouldn't because we do not have anything to return. We are actually just using the values from the provided `Context`.
`setSpanName()` I think this method is redundant to set the span name as we already use the provided context for that. I see it is only used in [RestProxy here](https://github.com/Azure/azure-sdk-for-java/blob/e2a19ea5db4b63c953948e41f0e25343faf0a221/sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java#L185).